internal/insights/store/store.go

```diff
@@ -976,11 +972,34 @@ func (s *Store) GetAllDataForInsightViewID(ctx context.Context, opts ExportOpts)
 		return nil, errors.Wrap(err, "fetching code insights data")
 	}
 
+	results = FilterSeriesPoints(denylist, results)
+
 	return results, nil
 }
 
+func FilterSeriesPoints(denyList []api.RepoID, array []SeriesPointForExport) []SeriesPointForExport {
+	// If there is nothing to filter, then return early. This skips the assignment in the later for-loop.
+	if len(denyList) == 0 {
+		return array
+	}
+	// We turn the denyList into a map to ensure O(n) time complexity
+	denyMap := make(map[int32]bool)
+	for _, repoID := range denyList {
+		denyMap[int32(repoID)] = true
+	}
```

Comment:
Go trick: we can use a zero-sized empty struct as the value to save some memory.

```suggestion
	denyMap := make(map[int32]struct{})
	for _, repoID := range denyList {
		denyMap[int32(repoID)] = struct{}{}
	}
```

Reply:
Also, why do we convert to `int32`? Can't se use `map[api.RepoID]struct{}` directly?

Reply:
Not sure why I used int32, probably misunderstood an error. It's now using api.RepoID across the board.